sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -48,203 +48,239 @@ public class ImplementIDisposableCorrectly : SonarDiagnosticAnalyzer
 
         protected sealed override void Initialize(SonarAnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(CheckIDisposableImplementation, SyntaxKind.ClassDeclaration);
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var checker = new DisposableChecker((ClassDeclarationSyntax)c.Node, c.SemanticModel);
+
+                if (!checker.IsCompliant())

```

Comment:
I am not really expecting a method to start with _Is_ it feels more like a property.